app/models/event.rb

```diff
@@ -29,10 +30,9 @@ class Event < ApplicationRecord
 
   scope :active, -> { where('deadline >= ?', Date.current) }
 
-  validates :name, :discipline, :game_mode, :player_type, presence: true
-  validates :deadline, :startdate, :enddate, presence: true
+  validates :name, :discipline, :game_mode,  presence: true
   validates :max_teams, numericality: { greater_than_or_equal_to: 0, allow_nil: true }
-  validate :end_after_start, :start_after_deadline
+  validate :end_after_start, :start_after_deadline #in tournament & league?
```

Comment:
I think moving this validation and the one for max_teams to tournaments and leagues is the best we can do.  
Sure code duplication might not be super pretty but it's probably still cleaner than having a type check in the event model.